src/bulls_and_cows_lib/board.cpp

```diff
@@ -3,6 +3,252 @@
+				const auto& feedback = attempt_and_feedback.feedback;
+				output_stream << "Attempt #" << (index + 1) << " = " << attempt.value << " | Bulls = " << feedback.bulls
+					<< " | Cows = " << feedback.cows << "\n";
+			}
+		}*/
+
+        // start of head printing
+        int anssize = game_options.number_of_characters_per_code;
+        std::string espaces = "";
+        std::string moins = "--------------------------";
+        std::string etoiles = "";
+        std::string points = "";
+
+        for (unsigned int i = 0U; i < game_options.number_of_characters_per_code; i++)
+        {
+            points = points + ". ";
+        }
+        for (int j = 0; j < anssize; j++)
+        {
+            etoiles = etoiles + "* ";
+            espaces = espaces + "  ";
+            moins = moins + "--";
+        }
+        moins = moins + "\n";
+
+        output_stream << moins << "| SECRET  " << etoiles << "|              |\n"
+                      << moins << "| ATTEMPTS" << espaces << "| BULLS | COWS |\n"
+                      << moins << "\n";
+        // end of head printing
+
+        auto sz = board.attempts_and_feedbacks.size();
+        for (unsigned int i = 0U; i < game_options.max_number_of_attempts; i++) // for all attempts
+        {
+            // board.attempts_and_feedbacks[sz - i];
+            if (i < game_options.max_number_of_attempts - sz) // writting the empty attempts
+            {
+
+                if (game_options.max_number_of_attempts - i < 10)
+                {
+                    output_stream << "| #" << '0' << game_options.max_number_of_attempts - i << '   ' << '   ' << points
+                                  << "|"
+                                  << "   "
+                                  << " "
+                                  << "   "
+                                  << "|"
+                                  << "  " << board.attempts_and_feedbacks[i].feedback.cows << "   |\n";
+                }
+                // two different cases to put the 0 digit in case of need
+                else
+                {
+                    output_stream << "| #" << game_options.max_number_of_attempts - i << '   ' << '   ' << points << "|"
+                                  << "   "
+                                  << " "
+                                  << "   "
+                                  << "|"
+                                  << "  "
+                                  << " "
+                                  << "   |\n";
+                }
+            }
+
+            else
+            { 
+				// writting played attempts
+                std::string codeout;
+                for (unsigned int k = 0U; k < game_options.number_of_characters_per_code; k++)
+                {
+                    codeout = codeout + board.attempts_and_feedbacks[i].attempt.value + " ";
+                }
+                if (game_options.max_number_of_attempts - i < 10)
```

Comment:
ici aussi le corps du if et du else semblent similaires et pourraient etre mutualisés dans une fonction parametrée

Reply:
Done. Deleted due to code changes.